### DISCUSSION TITLE
[HxGrid] Adding attributes on TR

### DISCUSSION THREAD

**Discussion by TPIvan** [havit/Havit.Blazor#922] [Q&A] [answered]
I am working on a feature to reorder rows in HxGrid by dragging. To achieve this, I am currently using a draggable cell content:
              
                <ItemTemplate Context="item">
                    <div @ondragstart="(e) => HandleDragStart(item, e)"
                         @ondrop="() => HandleDrop(item)"
                         @ondragenter="(e) => HandleDragEnter(item, e)"
                         @ondragleave="HandleDragLeave"
                         @ondragend="HandleDragEnd"
                         draggable="true"
                         ondragover="event.preventDefault();">
                        <HxIcon Icon="BootstrapIcon.GripVertical" />
                    </div>
                </ItemTemplate> 
For a better user experience, it would be ideal to enable dragging directly on the <tr> elements of the grid. This requires adding the necessary draggable attributes to the [<tr>](https://github.com/havit/Havit.Blazor/blob/ef209f3cec019c5ad2bed2b7cde2bdb46d941aa3/Havit.Blazor.Components.Web.Bootstrap/Grids/HxGrid.razor#L156) elements of the grid. 

I believe that adding draggable attributes might be too specialized. Therefore, I propose creating the ability to add or splat any attribute to the <tr> elements. This would provide greater flexibility for various use cases.

I am more than willing to contribute to the implementation, but I think designing a proper API for this feature is the most crucial part.

**Comment by hakenr** [comment]
Good idea, but...
Do you have a specific component in mind where you'd like the attributes to be splatted onto the `tr` element? 🤔

**Reply by TPIvan** [reply]
I was thinking of placing this functionality within the `Columns `element, as it serves as the container for `HxGridColumn `which becomes the cells (`<td>`). However, I'm not sure if this approach is feasible.

**Reply by hakenr** [reply]
`Columns` is a `RenderFragment` parameter and doesn't accept any attributes other than `Context` for the contextual parameter's name. To make `Columns` a component within a `ChildContent` parameter of `HxGrid`, we would have to completely restructure the `HxGrid` setup, which could also impact performance.

We need to find an alternative way to handle these attributes. One option might be using a `Dictionary<string, object>` elsewhere, like `<HxGrid TableRowAdditionalAttributes="..." />`. This approach may not be the most convenient for setting parameters, but it could be acceptable for those rare scenarios.

**Comment by TPIvan** [comment]
Using TableRowAdditionalAttributes is absolutely acceptable for me. The idea of splatting attributes from Columns was just too nice to be true. :-).

**Comment by hakenr** [comment, marked as answer]
Tracked as
- #923
